workspace_tools/targets.py

```diff
@@ -1018,29 +1018,65 @@ def binary_hook(t_self, resources, elf, binf):
 
             
 # 16KB MCU version, e.g. Nordic nRF51822, Seeed Arch BLE, etc.
-class MCU_NRF51_16K(MCU_NRF51):
+class MCU_NRF51_16K_BASE(MCU_NRF51):
     def __init__(self):
         MCU_NRF51.__init__(self)
         self.extra_labels += ['MCU_NORDIC_16K', 'MCU_NRF51_16K']
         self.macros += ['TARGET_MCU_NORDIC_16K', 'TARGET_MCU_NRF51_16K']
 
 # derivative class used to create softdevice+bootloader enabled images
-class MCU_NRF51_16K_BOOT(MCU_NRF51_16K):
+class MCU_NRF51_16K_BOOT_BASE(MCU_NRF51_16K_BASE):
     def __init__(self):
-        MCU_NRF51_16K.__init__(self)
+        MCU_NRF51_16K_BASE.__init__(self)
         self.extra_labels += ['MCU_NRF51_16K_BOOT']
         self.macros += ['TARGET_MCU_NRF51_16K_BOOT', 'TARGET_OTA_ENABLED']
         self.MERGE_SOFT_DEVICE = True
         self.MERGE_BOOTLOADER = True
 
 # derivative class used to create program only images for use with FOTA
-class MCU_NRF51_16K_OTA(MCU_NRF51_16K):
+class MCU_NRF51_16K_OTA_BASE(MCU_NRF51_16K_BASE):
     def __init__(self):
-        MCU_NRF51_16K.__init__(self)
+        MCU_NRF51_16K_BASE.__init__(self)
         self.extra_labels += ['MCU_NRF51_16K_OTA']
         self.macros += ['TARGET_MCU_NRF51_16K_OTA', 'TARGET_OTA_ENABLED']
         self.MERGE_SOFT_DEVICE = False
 
+class MCU_NRF51_16K(MCU_NRF51_16K_BASE):
+    def __init__(self):
```

Comment:
I am starting to be confused about so many mcu_nrf51 derivatives. we have multiple classes (MCU_NRF51, MCU_NRF51_size_BASE, MCU_NRF51_size_bootloaderversion.

Why all these new classes are needed?